cmd/publishing-bot/main.go

```diff
@@ -195,12 +198,24 @@ func main() { //nolint: gocyclo
 			}
 		} else {
 			// run
-			if _, _, err := publisher.Run(); err != nil {
-				glog.Infof("Failed to run publisher: %v", err)
+			if _, _, publisherErr = publisher.Run(); err != nil {
+				glog.Infof("Failed to run publisher: %v", publisherErr)
 			}
 		}
 
 		if *interval == 0 {
+			// This condition is specifically used by the CI to get the exit code
+			// of the bot on an unsuccessful run.
+			//
+			// In production, the bot will not exit with a non-zero exit code, since
+			// the interval will be always non-zero
+			if publisherErr != nil {
+				if exitErr, ok := publisherErr.(*exec.ExitError); ok {
+					os.Exit(exitErr.ExitCode())
+				} else {
+					os.Exit(1)
```

Comment:
Unix exit codes use `0` for success :)

https://tldp.org/LDP/abs/html/exit-status.html#:~:text=%23%20By%20convention%2C%20an%20'exit,an%20error%20or%20anomalous%20condition.

Reply:
I think `1` is meant as a generic exit code here, in case we encounter an error that doesn't come from exec-ing a process, ex: [failure to initialise logger](https://github.com/kubernetes/publishing-bot/blob/b30994a756b78e1bc65ba085061308413600d2b2/cmd/publishing-bot/publisher_logger.go#L43).

I'm assuming we'd want the presubmit to fail when things like that fail

Reply:
Nice thanks!